tests/isolation_provider/test_qubes.py

```diff
@@ -0,0 +1,117 @@
+import signal
+import subprocess
+import time
+
+import pytest
+from pytest import MonkeyPatch
+from pytest_mock import MockerFixture
+
+from dangerzone.conversion import errors
+from dangerzone.document import Document
+from dangerzone.isolation_provider.base import IsolationProvider
+from dangerzone.isolation_provider.qubes import Qubes, running_on_qubes
+
+# XXX Fixtures used in abstract Test class need to be imported regardless
+from .. import (
+    pdf_11k_pages,
+    sample_bad_height,
+    sample_bad_width,
+    sample_doc,
+    sanitized_text,
+    uncommon_text,
+)
+from .base import TestIsolationProvider
+
+
+@pytest.fixture
+def provider() -> Qubes:
+    return Qubes()
+
+
+@pytest.mark.skipif(not running_on_qubes(), reason="Not on a Qubes system")
+class TestQubes(TestIsolationProvider):
+    def test_max_pages_client_side_enforcement(
+        self,
+        sample_doc: str,
+        provider: Qubes,
+        mocker: MockerFixture,
+    ) -> None:
+        provider.progress_callback = mocker.MagicMock()
+        mocker.patch(
+            "dangerzone.conversion.errors.MAX_PAGES", 1
+        )  # sample_doc has 4 pages > 1
+        doc = Document(sample_doc)
+        with pytest.raises(errors.MaxPagesException):
+            success = provider._convert(doc, ocr_lang=None)
+            assert not success
+
+    def test_max_dimentions(
+        self,
+        sample_bad_width: str,
+        sample_bad_height: str,
+        provider: Qubes,
+        mocker: MockerFixture,
+    ) -> None:
+        provider.progress_callback = mocker.MagicMock()
+        with pytest.raises(errors.MaxPageWidthException):
+            success = provider._convert(Document(sample_bad_width), ocr_lang=None)
+            assert not success
+        with pytest.raises(errors.MaxPageHeightException):
+            success = provider._convert(Document(sample_bad_height), ocr_lang=None)
+            assert not success
+
+    def test_inconsistent_dimentions(
+        self,
+        provider: Qubes,
+        mocker: MockerFixture,
+        monkeypatch: MonkeyPatch,
+        sample_doc: str,
+    ) -> None:
+        provider.progress_callback = mocker.MagicMock()
+
+        def qrexec_subprocess() -> subprocess.Popen:
+            p = subprocess.Popen(
+                ["tee"],  # pipe out whatever we feed into stdin
+                stdin=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            assert p.stdin is not None
+            p.stdin.write(b"\x00\x01")  # 1 page
+            p.stdin.write(b"\x00\x04")  # 4px width
+            p.stdin.write(b"\x00\x05")  # 5px height
+            p.stdin.write(b"\x00")  # only one byte received but 3*4*5 expected
+            return p
```

Comment:
This test now fails in my machine, due to the PRs that have just been merged. The reason is that `tee` does not exit, so the client-side will not receive EOF. It will fail with a timeout error instead, not `errors.InterruptedConversion`.

Reply:
Hm, I think that `tee & exit` now makes the test racy. What if `tee` exits before it sends back the input? I think a better way is to change `tee & exit` with `echo -e`:

```diff
diff --git a/tests/isolation_provider/test_qubes.py b/tests/isolation_provider/test_qubes.py
index a1e5efb..f3ab918 100644
--- a/tests/isolation_provider/test_qubes.py
+++ b/tests/isolation_provider/test_qubes.py
@@ -71,17 +71,14 @@ class TestQubes(TestIsolationProvider):
 
         def qrexec_subprocess() -> subprocess.Popen:
             p = subprocess.Popen(
-                ["tee & exit"],  # pipe out whatever we feed into stdin
+                ["echo",
+                 "-n",
+                 "-e",
+                 r"\x00\x01\x00\x04\x00\x05\x99"],
                 stdin=subprocess.PIPE,
                 stdout=subprocess.PIPE,
                 stderr=subprocess.PIPE,
-                shell=True,
             )
-            assert p.stdin is not None
-            p.stdin.write(b"\x00\x01")  # 1 page
-            p.stdin.write(b"\x00\x04")  # 4px width
-            p.stdin.write(b"\x00\x05")  # 5px height
-            p.stdin.write(b"\x00")  # only one byte received but 3*4*5 expected
```

Reply:
That also works. Thanks for the suggestions. Implemented in 0f189c1d.